tensor_gc.go

```diff
@@ -13,20 +13,19 @@ import (
 
 var (
 	tensorFinalizersWG = &sync.WaitGroup{}
-	gcPrepared         = false
 )
 
 // SetTensorFinalizer sets a finalizer to the tensor
 func SetTensorFinalizer(t *unsafe.Pointer) {
 	// We don't want the following conditional and the finalizer using
 	// different gcPrepared values, so we leverage p and closure here.
-	p := gcPrepared
-	if p {
+	p := C.GCPrepared()
```

Comment:
Do we need a lock when calling C.GCPrepared() or in that function?

Reply:
No, as explained above. `C.GCPrepared` returns a copy of the`thread_local`.